src/test/java/org/minions/devfund/katerinaanzoleaga/battleship/ShipTest.java

```diff
@@ -0,0 +1,280 @@
+package org.minions.devfund.katerinaanzoleaga.battleship;
+
+import junitparams.Parameters;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+//import org.junit.experimental.categories.Category;
+import junitparams.JUnitParamsRunner;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Ship test class.
+ */
+@RunWith(JUnitParamsRunner.class)
+public class ShipTest {
+    private BattleShip battleShip = new BattleShip();
+    private BattleCruiser battleCruiser = new BattleCruiser();
+    private Cruiser cruiser = new Cruiser();
+    private LightCruiser lightCruiser = new LightCruiser();
+    private Destroyer destroyer = new Destroyer();
+    private Submarine submarine = new Submarine();
+    private EmptySea emptySea = new EmptySea();
+
+    private static Ocean theOcean = new Ocean();
+
+    /**
+     * Parametrized test to test PacingShips in some valid and not valid positions.
+     * @param row int
+     * @param column int
+     * @param horizontal bool
+     * @param expectedResult bool
+     */
+    @Test
+    @Parameters({
+            "0, 8, true, false",
+            "0, 0, false, false",
+            "1, 5, false, false ",
+            "19,8, true, true",
+            "-1, 22, true, false",
+            "19, 18, true, false"
+    })
+    public void testOkToPlaceShipAt(int row, int column, boolean horizontal, boolean expectedResult) {
+        battleShip.placeShipAt(0, 0, true, theOcean);
+        assertEquals(expectedResult, battleShip.okToPlaceShipAt(row, column, horizontal, theOcean));
+    }
+
+    /**
+     * Test shot at different positions of a battleCruiser.
+     */
+    @Test
+    public void testShootAt() {
+        final int row = 0;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "row" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)